### PR TITLE
Add random seed arg

### LINKED ISSUE #944: Random seed arg
We currently have a hardcoded random seed here:

https://github.com/Lightning-AI/lit-gpt/blob/44c3c58b759fa0903ab31ed8863a66c157d5ccd9/finetune/full.py#L120

I think we should make this a CLI arg. What do you think @carmocca ?

### PR DESCRIPTION
**Pull request by rasbt**
Exposes the random seed as a configuration for the CLI. 

Fixes #944

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]
Good idea. Maybe the default should be 0 rather than a funny number like 1337? I think this default came from nano-gpt.

**Comment by carmocca** [comment]
Why not put it inside one of the args classes?

**Comment by rasbt** [comment]
I was thinking that this is more of a global setting that currently affects both training and evaluation. E.g., we have a `randint` in `get_batch` for getting a validation batch. It would probably change with the dataloader refactoring, but in general, I was thinking we might potentially add some evaluation in the future that might be affected by a global random seed (just to leave that door open).

But in any case I'm happy to make it a `TrainArg` if you think it's a better fit.
